src/components/tasks/TaskStatusDropdown.tsx

```diff
@@ -29,20 +30,29 @@ export default function TaskDropDown({
         BACKEND_TASK_STATUS.DONE = BACKEND_TASK_STATUS.COMPLETED;
     }
 
-    const taskStatus = Object.entries(BACKEND_TASK_STATUS).filter(
-        ([key]) =>
-            !(key === BACKEND_TASK_STATUS.COMPLETED) &&
-            !(!isDevMode && key === BACKEND_TASK_STATUS.BACKLOG)
-    );
+    const getAvailableTaskStatuses = () => {
+        const excludedStatuses = [
+            BACKEND_TASK_STATUS.COMPLETED,
+            BACKEND_TASK_STATUS.BACKLOG,
+            BACKEND_TASK_STATUS.MERGED,
+            BACKEND_TASK_STATUS.UN_ASSIGNED,
+        ];
```

Comment:
- move this outside of the function to prevent the array from being recreated on function call.